frontend/packages/ux-editor/src/components/config/EditFormComponent.tsx

```diff
@@ -50,31 +52,37 @@ export const EditFormComponent = ({
     return componentSpecificEditConfig[component.type];
   };
 
-  const toggleShowBetaFunc = () => {
-    setShowBetaFunc(!showBetaFunc);
+  const toggleShowBetaFunc = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setShowComponentConfigBeta(event.target.checked);
+    // Ensure choice of feature toggling is persisted in local storage
+    if(event.target.checked){
+      addFeatureFlagToLocalStorage('componentConfigBeta');
+    } else {
+      removeFeatureFlagFromLocalStorage('componentConfigBeta');
+    }
   };
 
   return (
     <LegacyFieldSet className={classes.root}>
       <LegacyCheckbox
         onChange={toggleShowBetaFunc}
-        checked={showBetaFunc}
+        checked={showComponentConfigBeta}
         label={t('ux_editor.edit_component.show_beta_func')}
         helpText={t('ux_editor.edit_component.show_beta_func_helptext')}
       />
       <Heading level={2} size='xsmall'>
         {getComponentTitleByComponentType(component.type, t)} ({component.type})
       </Heading>
-      {showBetaFunc && isLoading && <AltinnSpinner spinnerText={t('general.loading')} />}
-      {showBetaFunc && (
+      {showComponentConfigBeta && isLoading && <AltinnSpinner spinnerText={ t('general.loading') } />}
+      {showComponentConfigBeta && (
         <FormComponentConfig
           schema={isLoading ? {} : schema}
           component={component}
           editFormId={editFormId}
           handleComponentUpdate={handleComponentUpdate}
         />
       )}
-      {!showBetaFunc && (
+      {!showComponentConfigBeta && (
```

Comment:
I think we should check for isLoading on here also :)

Reply:
I don't think it's needed here. The data that is loaded is only relevant for the new view, and not for the old one.